usecases/scheduler/src/lambda/ssv2_configUpdater/configUpdater.py

```diff
@@ -0,0 +1,274 @@
+                print('Unable to remove {}, current status is PendingConfirmation. Amazon will delete automatically the subscription after 3 days of creation'.format(existingEmail.get('Endpoint')))
+                continue
+
+            print('Removing {}'.format(existingEmail.get('Endpoint')))
+            sns.unsubscribe(SubscriptionArn=existingEmail.get('SubscriptionArn'))
+
+        if existingEmail.get('Endpoint') in emails:
+            emails.remove(existingEmail.get('Endpoint'))
+
+    for email in emails:
+        print('Adding {}'.format(email))
+        sns.subscribe(
+            TopicArn=snsArn,
+            Protocol='email',
+            Endpoint=email
+        )
+    
+    pass    
+
+def deleteEventBridge(configId):
+    print("Attempting to delete EventBridge Scheduler: ...")
+
+    try:
+        scheduler.delete_schedule(
+        GroupName=schedulerGroupName,
+        Name='ScreenerScheduler-' + configId
+        )
+        return True
+    except botocore.exceptions.ClientError as e:
+        print(e.response['Error']['Message'])
+        return False
+
+def deleteSnsRecipient(ssv2configId):
+    # List all topics to find the one with the desired name
+    topics = []
+    next_token = None
+    topic_name = snsArnPrefix + '-' + ssv2configId
+    
+    while True:
+        if next_token:
+            response = sns.list_topics(NextToken=next_token)
+        else:
+            response = sns.list_topics()
+        
+        topics.extend(response['Topics'])
+        
+        if 'NextToken' in response:
+            next_token = response['NextToken']
+        else:
+            break
+    
+    # Find the topic with the specified name
+    topic_arn = None
+    for topic in topics:
+        if topic_name in topic['TopicArn']:
+            topic_arn = topic['TopicArn']
+            break
+    
+    if topic_arn is not None:
+        # Delete the topic by its ARN
+        sns.delete_topic(TopicArn=topic_arn)
+        print(f"Deleted topic: {topic_arn}")
+        return True
+    else:
+        print(f"Topic with name '{topic_name}' not found.")
+        return False
+
+
+
+
+def sanitizeEvent(event, action):
+    records = event['Records']
+    deserializer = ddbDeserializer()
+
+    items = [] 
+
+    if action == 'INSERT':
+        for record in records:
+            configId = record['dynamodb']['Keys']['name']['S']
+            _sanitized = {k: deserializer.deserialize(v) for k,v in record['dynamodb']['NewImage'].items()}
+            
+            params = ''
+            regions = list(_sanitized['regions'])
+            services = list(_sanitized['services'])
+            
+            if regions == ['']:
+                regions = ['ALL']
+            
+            if services == ['']:
+                services = None
+
+            params = "--regions {}".format( ','.join(regions) )
+            if services:
+                params = params + " --services {}".format( ','.join(services))
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that you are concatenating a string inside a loop. Concatenating immutable sequences will always results in a new object. This means that building up a sequence by repeated concatenation will have a quadratic runtime cost in the total sequence length. For efficiency and to get a linear runtime cost, consider appending values to a list and concatenating that list using [str.join()](https://docs.python.org/3/library/stdtypes.html#str.join) or [bytes.join()](https://docs.python.org/3/library/stdtypes.html#bytes.join) method.

[Learn more](https://docs.python.org/3/faq/programming.html#what-is-the-most-efficient-way-to-concatenate-many-strings-together)